cmd/descheduler/app/server.go

```diff
@@ -79,6 +80,16 @@ func NewDeschedulerCommand(out io.Writer) *cobra.Command {
 				}
 			}
 
+			// healthz returns true when health check pass
+			var checker []healthz.HealthChecker
+			checker = append(checker,  healthz.NamedCheck("Descheduler", healthz.PingHealthz.Check))
+			newMux := mux.NewPathRecorderMux("healthz")
+			healthz.InstallHandler(newMux, checker...)
+
+			if _, err := SecureServing.Serve(newMux, 0, ctx.Done()); err != nil {
```

Comment:
Unfortunately, this does not work for me. Though this does:
```
diff --git a/cmd/descheduler/app/server.go b/cmd/descheduler/app/server.go
index 555beaf41..129756c3f 100644
--- a/cmd/descheduler/app/server.go
+++ b/cmd/descheduler/app/server.go
@@ -20,6 +20,7 @@ package app
 import (
        "context"
        "io"
+
        "k8s.io/apiserver/pkg/server/healthz"
 
        "sigs.k8s.io/descheduler/cmd/descheduler/app/options"
@@ -70,24 +71,19 @@ func NewDeschedulerCommand(out io.Writer) *cobra.Command {
                        }
 
                        ctx := context.TODO()
+                       pathRecorderMux := mux.NewPathRecorderMux("descheduler")
                        if !s.DisableMetrics {
-                               pathRecorderMux := mux.NewPathRecorderMux("descheduler")
                                pathRecorderMux.Handle("/metrics", legacyregistry.HandlerWithReset())
-
-                               if _, err := SecureServing.Serve(pathRecorderMux, 0, ctx.Done()); err != nil {
-                                       klog.Fatalf("failed to start secure server: %v", err)
-                                       return
-                               }
                        }
 
                        // healthz returns true when health check pass
                        var checker []healthz.HealthChecker
-                       checker = append(checker,  healthz.NamedCheck("Descheduler", healthz.PingHealthz.Check))
-                       newMux := mux.NewPathRecorderMux("healthz")
-                       healthz.InstallHandler(newMux, checker...)
+                       checker = append(checker, healthz.NamedCheck("Descheduler", healthz.PingHealthz.Check))
+                       healthz.InstallHandler(pathRecorderMux, checker...)
 
-                       if _, err := SecureServing.Serve(newMux, 0, ctx.Done()); err != nil {
+                       if _, err := SecureServing.Serve(pathRecorderMux, 0, ctx.Done()); err != nil {
                                klog.Fatalf("failed to start secure server: %v", err)
+                               return
                        }
 
                        err := Run(s)
```

Maybe creating two muxes is not the right way to go.

Reply:
Yes, you are right, so sorry for I don't found it, I will fix it soon.